docs/config-v3-documentation.md

```diff
@@ -0,0 +1,953 @@
+<!-- md:default @auto -->
+
+String describing result on each test. It should have length of the number of tests (including [`[test00]`](#test00)).
+
+Recognized results are:
+
+- `1` — success
+- `0` — fail
+- `P` — partial success
+- `S` — superoptimal success
+- `A` — success / superoptimal success
+- `W` — wrong answer
+- `!` — runtime error
+- `T` — timeout
+- `N` — normalization fail
+- `X` — any result
+
+The result of a test is the result of the worst testcase.
+`W!TN` require at least one testcase with the corresponding verdict.
+
+- `@all` — string of `11...`
+- `@any` — string of `XX...`
+- `@auto` — `@all` if this is primary solution, `@any` otherwise
+
+### points
+<!-- md:version 1.0.0 -->
+<!-- md:type int -->
+<!-- md:default X -->
+
+Points that program should get or `X` for any number of points.
+
+### points_min
+<!-- md:version 1.7.0 -->
+<!-- md:type int | 'X' -->
+<!-- md:default X -->
+
+### points_max
+<!-- md:version 1.7.0 -->
+<!-- md:type int | 'X' -->
+<!-- md:default X -->
+
+Upper and lower bounds on points.
+
+Cannot be set simultaneously with [`points`](#points).
+
+## [solution_correct]
+
+Section for specific solution. This one is named `correct`.
+
+!!! example Task with few solutions
+    For a task with samples, two other tests and solutions `correct.py`, `wrong.cpp` and `slow.java`
+    config might look:
+    ```ini
+    [solution_correct]
+    primary=yes
+    # primary solution must pass on all tests
+
+    [solution_wrong]
+    # we don't really care about samples, otherwise wrong
+    tests=XWW
+
+    [solution_slow]
+    # timeout on test02, otherwise correct
+    tests=11T
+    ```
+
+
+## [run]
+Sections describing running a program.
+
+The run section for each program is optional. If it is missing, its contents are autogenerated,
+by looking for most specific section with the given key set, according to this hierarchy:
+```
+                                         [run]
+      [run_gen]          [run_validator]          [run_judge]                         [run_solution]
+                                                                      [run_primary_solution] [run_secondary_solution]
+[run_gen:{program}] [run_validator:{program}] [run_judge:{program}]              [run_solution:{program}]
+```
+
+!!! note
+    Usual tasks have only few `[run]` sections:
+    ```ini
+    [run_solution]
+    time_limit=1
+    subdir=solutions
+
+    [run_judge]
+    time_limit=2
+    ```
+
+### build {#run-build}
+<!-- md:version 2.0.0 -->
+<!-- md:type build_ref -->
+<!-- md:default @auto -->
+
+Reference to the build section specifying how to build this program.
+
+`@auto` expands to `{program_role}:{subdir}/{program}`, e.g. `solution:solutions/solve`.
+
+See [`[build]`](#build) for more
+
+### exec
+<!-- md:version 1.6.0 -->
+<!-- md:type path -->
+<!-- md:default-empty -->
+
+Filename of the program to execute relative to the built directory.
+If a directory is chosen directory/run is executed instead.
```

Comment:
I'm not sure it's clear "directory/run" is supposed to refer to a path when it's not monospaced.

```suggestion
If a directory is chosen `{exec}/run` is executed instead.
```